spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionInfoService.java

```diff
@@ -127,6 +127,8 @@ public TaskExecutionInformation findTaskExecutionInformation(String taskName,
 
 		TaskDefinition originalTaskDefinition = taskDefinitionRepository.findById(taskName)
 				.orElseThrow(() -> new NoSuchTaskDefinitionException(taskName));
+		//TODO: This normally called by JPA automatically but `AutoCreateTaskDefinitionTests` fails without this.
```

Comment:
Do we want to go forward with the initialize here or is it safe to use `@Transactional(propagation = Propagation.SUPPORTS)`?   I know we discussed not using Propagation.NOT_SUPPORTED.

Reply:
AFAIK `propogation.SUPPORTS` does not create a transaction but will participate if already running in one. I don't think that will work here.

Reply:
👍